### DISCUSSION TITLE
Forward Event Cycle detection

### DISCUSSION THREAD

**Discussion by dontworkagain** [Netflix/conductor#2938] [Q&A]
Hi guys,
Can anyone tell me if it's possible to "Prevent event forwarding to the beginning of the cycle."?

**Comment by v1r3n** [comment]
Hi @dontworkagain can you provide more context on the question?

**Reply by dontworkagain** [reply]
Hi, can you please tell me if Conductor has the possibility to "Forward the received event to another event system if the event reaches a certain condition (such as date and time)?
Because if possible, I would like to know if I could prevent a sequential forwarding of events to return to the beginning of the sequence.
Something like to know if a graph is a cycle. (There is a cycle in a graph only if there is a back edge present in the graph. )

**Reply by apanicker-nflx** [reply]
You could build event cycle detection as part of your workflow by implementing the first task in your workflow as the loop checker, akin to the exit criteria in a recursive function. There is no explicit support for event cycle detection but this can easily be added as a task within your workflow.

**Reply by dontworkagain** [reply]
Hey. Thank you for your help. I think this will be useful.
